lib/mediakit_player.dart

```diff
@@ -158,60 +211,76 @@ class MediaKitPlayer extends AudioPlayerPlatform {
       _dataController.stream;
 
   /// Updates the playback event
-  void _updatePlaybackEvent(
-      {Duration? duration, IcyMetadataMessage? icyMetadata}) {
+  void _updatePlaybackEvent() {
     _eventController.add(PlaybackEventMessage(
       processingState: _processingState,
       updateTime: DateTime.now(),
       updatePosition: _position,
       bufferedPosition: _bufferedPosition,
-      duration: duration,
-      icyMetadata: icyMetadata,
+      duration: _duration,
+      icyMetadata: null,
       currentIndex: _currentIndex,
       androidAudioSessionId: null,
+      errorCode: _errorCode,
+      errorMessage: _errorMessage,
     ));
   }
 
   @override
   Future<LoadResponse> load(LoadRequest request) async {
     _logger.finest('load(${request.toMap()})');
+    _mediaOpened = false;
+    _loadCompleter = Completer();
     _currentIndex = request.initialIndex ?? 0;
     _bufferedPosition = Duration.zero;
     _position = Duration.zero;
-
-    _processingState = ProcessingStateMessage.buffering;
+    _duration = null;
+    _processingState = ProcessingStateMessage.loading;
+    _errorCode = null;
+    _errorMessage = null;
+    _updatePlaybackEvent();
 
     if (request.audioSourceMessage is ConcatenatingAudioSourceMessage) {
       final audioSource =
           request.audioSourceMessage as ConcatenatingAudioSourceMessage;
-      final playable = Playlist(
+      final playable = _playable = Playlist(
```

Comment:
Could we not avoid keeping `_playable` as a class property since we have `_mediaOpened` that keeps track if a media was loaded or not?

Reply:
Yes, you're right. I intended to get rid of that variable before committing, and was also wondering if there was some state inside the media_kit player that I could use so I could get rid of both variables, but didn't find any. I'll fix this.